js/libs/utils.js

```
@@ -0,0 +1,275 @@
+    return new Date(obj.getTime());
+  }
+
+  if (obj instanceof Array) {
+    return obj.map(item => cloneDeep(item));
+  }
+
+  if (typeof obj === 'object') {
+    const cloned = {};
+
+    for (const key in obj) {
+      if (obj.hasOwnProperty(key)) {
```

Biome (1.9.4)

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)